src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.cs

```diff
@@ -12,74 +12,81 @@ namespace Microsoft.Identity.Abstractions
-        /// <see cref="DownstreamRestApiOptions.HttpMethod"/> is <code>HttpMethod.Patch</code>, 
+        /// <see cref="AuthorizationHeaderProviderOptions.HttpMethod"/> is <code>HttpMethod.Patch</code>, 
         /// <see cref="HttpMethod.Post"/>, <see cref="HttpMethod.Put"/>.</param>
         /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>An <see cref="HttpResponseMessage"/> that the application will process.</returns>
-        Task<HttpResponseMessage> CallRestApiForAppAsync(
+        Task<HttpResponseMessage> CallRestApiForUserAsync(
             string serviceName,
-            Action<DownstreamRestApiOptions>? DownstreamRestApiOptionsOverride = null,
+            Action<DownstreamRestApiOptions>? downstreamRestApiOptionsOverride = null,
+            ClaimsPrincipal? user = default,
             HttpContent? content = null,
             CancellationToken cancellationToken = default);
 
         /// <summary>
-        /// Get a strongly typed response from the web API.
+        /// Calls the downstream REST API on behalf of the app itself, with the required scopes.
         /// </summary>
-        /// <typeparam name="TOutput">Output type.</typeparam>
         /// <param name="serviceName">Name of the service describing the downstream REST API. There can
         /// be several configuration named sections mapped to a <see cref="DownstreamRestApiOptions"/>,
-        /// each for one downstream REST API. You can pass-in null, but in that case <paramref name="DownstreamRestApiOptionsOverride"/>
+        /// each for one downstream REST API. You can pass-in null, but in that case <paramref name="downstreamRestApiOptionsOverride"/>
         /// needs to be set.</param>
-        /// <param name="DownstreamRestApiOptionsOverride">Overrides the options proposed in the configuration described
+        /// <param name="downstreamRestApiOptionsOverride">(Optional) Overrides the options proposed in the configuration described
         /// by <paramref name="serviceName"/>.</param>
-        /// <param name="user">[Optional] Claims representing a user. This is useful in platforms like Blazor
-        /// or Azure Signal R, where the HttpContext is not available. In other platforms, the library
-        /// will find the user from the HttpContext.</param>
+        /// <param name="content">Content to send to the REST API in the case where
+        /// <see cref="AuthorizationHeaderProviderOptions.HttpMethod"/> is <code>HttpMethod.Patch</code>, 
+        /// <see cref="HttpMethod.Post"/>, <see cref="HttpMethod.Put"/>.</param>
         /// <param name="cancellationToken">Cancellation token.</param>
-        /// <returns>A strongly typed response from the web API.</returns>
-        Task<TOutput?> GetForUserAsync<TOutput>(
+        /// <returns>An <see cref="HttpResponseMessage"/> that the application will process.</returns>
+        Task<HttpResponseMessage> CallRestApiForAppAsync(
```

Comment:
There is no claims principal here, how will inbound artifacts be surfaced?

Reply:
It's calling on behalf of the apps. Would there be cases where the context is needed?

Reply:
Sometimes an application needs to create an outbound artifact using artifacts from the inbound request.
Api will require the implementation to put those inbound artifacts on the HttpContext or some other location.
If we were to pass a ClaimsPrincipal, we could attach those artifacts to the ClaimsPrincipal.ClaimsIdentity.BootstrapContext.